components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/config/model/graph/graaljs/GraalSerializableJsFunction.java

```diff
@@ -0,0 +1,147 @@
+/*
+ * Copyright (c) 2023, WSO2 LLC. (http://www.wso2.com).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.application.authentication.framework.config.model.graph.graaljs;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.graalvm.polyglot.Context;
+import org.graalvm.polyglot.PolyglotException;
+import org.graalvm.polyglot.Source;
+import org.graalvm.polyglot.Value;
+import org.wso2.carbon.identity.application.authentication.framework.config.model.graph.BaseSerializableJsFunction;
+
+import java.io.IOException;
+import java.util.function.Function;
+
+/**
+ * Javascript function wrapper. This allows serialization of a javascript defined function.
+ * Since Nashorn is deprecated in JDK 11 and onwards. We replaced it with GraalJS classes.
+ */
+public class GraalSerializableJsFunction implements BaseSerializableJsFunction<Context> {
+
+    private static final Log log = LogFactory.getLog(GraalSerializableJsFunction.class);
+    private static final long serialVersionUID = -7001351065432647040L;
+    private String source;
+    private boolean isPolyglotFunction = false;
+    private boolean isHostFunction = false;
```

Comment:
Check if we need to serialize the host function

Reply:
We don't need to serialize a host function. 
Addressed with f5809f2219be414b951ff3e9a92c5e56ec20d3ab